src/features/orientation/infrastructure/presentation/pages/localisation/localisation.page.ts

```diff
@@ -44,13 +48,31 @@ export class LocalisationPage {
   }
 
   public onGeoLocate(): void {
+    this._loadingState$.next(true);
     window.navigator.geolocation.getCurrentPosition((position: GeolocationPosition): void => {
       this.orientationLayout.filterForm.get('latitude')?.setValue(position.coords.latitude);
       this.orientationLayout.filterForm.get('longitude')?.setValue(position.coords.longitude);
       this.orientationLayout.filterForm.get('address')?.setValue(null);
+      this.getAddressFromCoordinates();
     });
   }
 
+  public _addressFromCoordinates$: BehaviorSubject<string> = new BehaviorSubject<string>('');
+  public addressFromCoordinates$: Observable<string> = this._addressFromCoordinates$.asObservable();
+
+  private _loadingState$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  public loadingState$: Observable<boolean> = this._loadingState$.asObservable();
+
+  public getAddressFromCoordinates(): void {
+    let url = `https://${BASE_NOMINATIM_URL}/reverse?format=json&lat=${this.orientationLayout.filterForm.value.latitude}&lon=${this.orientationLayout.filterForm.value.longitude}`;
+    this.http
+      .get(url)
+      .subscribe((data: any) =>
```

Comment:
On évite d'appeler la méthode `subscribe`, car elle oblige de faire appel à un état intermédiaire pour stocker le résultat, à la place on va plutôt utiliser `| async`.

Le problème c'est que parfois on ne sait pas à quelle donnée rattacher un `| async` dans le html, l'astuce pour trouver comment faire c'est de raisonner en termes de "flux" et se demander à quel "flux" on veut rattacher cette information.

Dans le cas présent on veut :
- Détecter que l'utilisateur a demandé une géolocalisation
- Mettre l'état du chargement à true
- Récupérer les coordonnées de manière asynchrone
- Faire un reverse géocode des coordonnées
- Mettre l'état du chargement à false

Il existe déjà un observable pour  l'état du chargement : `loadingState$`, on peut donc brancher les opérations qui ont lieu entre le moment ou le loader change d'état sur le flux.

Pour ce faire on a besoin de récupérer les coordonnées de manière asynchrone, c'est déjà le cas dans la méthode `onGeoLocate`, mais on ne peux pas fusionner avec l'observable `loadingState$` dans l'état, on peut donc créer le subkect `_geoLocation$` (`private _geoLocation$: Subject<Localisation> = new Subject<Localisation>();`), il suffit ensuite d'appeller `this._geoLocation$.next` dans le callback de `window.navigator.geolocation.getCurrentPosition` :
```typescript
this._geoLocation$.next(
  Localisation({
    latitude: position.coords.latitude,
    longitude: position.coords.longitude
  })
);
```
Maintenant il faut fusionner les deux observables, cela se fait de cette manière : 
```typescript
  public loadingState$: Observable<boolean> = this._loadingState$.pipe(
    mergeWith(
      this._geoLocation$
    )
  );
```
Le problème ici c'est que les deux observables ne gèrenet pas